pkg/graphgen/cycle.go

```diff
@@ -0,0 +1,52 @@
+package graphgen
+
+import (
+	"strconv"
+
+	"github.com/arangodb/feed/pkg/datagen"
+)
+
+type CycleGraphParameters struct {
+	Length uint64
+	Prefix string
+}
+
+func (c *CycleGraphParameters) MakeGraphGenerator() (GraphGenerator, error) {
```

Comment:
Same here, only create channel and go routine when somebody asks for the channel.